test/index.spec.ts

```diff
@@ -24,123 +26,171 @@ describe('ipns', function () {
     peerId = await peerIdFromKeys(rsa.public.bytes, rsa.bytes)
   })
 
-  it('should create an ipns record correctly', async () => {
+  it('should create an ipns record (V1+V2) correctly', async () => {
     const sequence = 0
     const validity = 1000000
 
-    const entry = await ipns.create(peerId, cid, sequence, validity)
-    expect(entry).to.deep.include({
+    const record = await ipns.create(peerId, cid, sequence, validity)
+
+    expect(record.value()).to.equal(uint8ArrayToString(cid))
```

Comment:
:warning: found interesting divergence here – unsure if this is just a bug in tests or a a legacy we should clean up across go/js and specs.

Seems that: 
- :green_circle:  GO creates records with `/ipfs/cid` paths
   -  GO  uses the same assumption in tests ([example](https://github.com/ipfs/boxo/blob/d4e6cd9eff8fb617d5dc52ef2a48fe10a3647b01/ipns/record_test.go#L48-L49)) 
- :green_circle:  The IPNS Specs only mention content paths (like `/ipfs/cid`) and do not even mention raw CIDs ([see "Value" definition here](https://specs.ipfs.tech/ipns/ipns-record/#ipns-record)).
- :question:  unsure what happens when `boxo/ipns` parses and validates a record that has  a raw CID bytes in the value field
   - in the past I've seen our libraries special-case this and upgrade detached CIDs to `/ipfs/Qm` content paths
- :anger:  JS tests here use raw CID bytes as string  but this looks like a bug
  -  `create` funtion takes `value: Uint8Array`  and it happens to be named "cid" here for some reason, which is imo a bug
  - Tests pass fine because nothing is validating the value to be a real content path, and if someone reads these tests as source of truth, they will implement IPNS Record incorrectly

I think at minimum, we should
- [ ] rename `cid=Qm..` to `contentPath=/ipfs/cid`  in all tests here
- [ ] add regression tests here (`js-ipns`) and in `boxo/ipns` that show what is the behavior when `Value` is a raw CID bytes vs a content path 
  - I would assume only content paths are valid and raw CIDs produce error, in such case we only need to update JS side to do the right thing and produce records with content paths.
  - If it happens that IPNS records with  raw CIDs instead of content paths were always allowed, we need to have test for both and also update "Value" description in ipns specs (no need for IPIP, as we fix spec to reflect  current state).

@hacdias does this sound sensible, or is there an easier way to clean this up?

Reply:
> I would assume only content paths are valid and raw CIDs produce error, in such case we only need to update JS side to do the right thing and produce records with content paths.

Similarly to what I said before, they are allowed in both:
- js-ipns does not make any verifications of what _value_ is. It can be a cat, or a dog, it doesn't matter. There is no path type as we have in Go.
- go-ipns will allow paths conforming to `path.Path`. In that case, it can be a path prefixed with a namespace, or a CID that is automatically prefixed with `/ipfs`.

With this being said, I'm not sure about how we should go from now on. We can indeed add tests to both implementations, but that won't help much: js-ipns let's you create a record with **anything** inside as far as I understand, while go-ipns is the one enforcing types. And yet, it supports self-standing CIDs that are not prefixed with the namespace.

Reply:
Hm.. seems this was missed because most of publishing of IPNS happens in GO, and JS path just happened to work due to liberal GO.

The priority would be to
1. not break existing JS users that pass raw CIDs OR resolve records that have raw CID instead of path in them.
2. make js-ipns produce IPNS Records conforming to the spec (which says this is always a content path)

My thinking out of this mess is to do something along these lines:
* creation: During IPNS Record creation we can check if passed string value starts with `/` and if not, check if it is a valid CID (and we need to check if it is either a base-encoded string or raw bytes as string), and if so, we convert it to a string with a content path  `/ipfs/cid` where cid is cidv1 base32 (or error otherwise saying a content path must start with `/`).
  * this way produced record always has content path and is compliant with spec, and people can pass content paths other than `/ipfs`
* verification: make `IPNSRecord.value()` do the same conversion (if protobuf/cbor value was raw cid, convert it to `/ipfs/cid` content path before returning to the user)

This way: 
* JS and GO produce the same records (with `/ipfs/cid` content paths)
* We don't break anything, Values end users interact with are always compatible with specs, and over time IPNS records expire and raw CIDs are no longer used.

Reply:
Done!